lz4/block/_block.c

```diff
@@ -199,27 +229,16 @@ compress (PyObject * Py_UNUSED (self), PyObject * args, PyObject * kwargs)
       dest_start = dest;
     }
 
-  switch (comp)
-    {
-    case DEFAULT:
-      output_size = LZ4_compress_default (source.buf, dest_start, source_size,
-                                          dest_size);
-      break;
-    case FAST:
-      output_size = LZ4_compress_fast (source.buf, dest_start, source_size,
-                                       dest_size, acceleration);
-      break;
-    case HIGH_COMPRESSION:
-      output_size = LZ4_compress_HC (source.buf, dest_start, source_size,
-                                     dest_size, compression);
-      break;
-    }
+  output_size = lz4_compress_generic (comp, source.buf, dest_start, source_size,
+				      dest_size, dict.buf, dict.len, acceleration,
+				      compression);
 
   Py_END_ALLOW_THREADS
 
   PyBuffer_Release(&source);
+  PyBuffer_Release(&dict);
 
-  if (output_size <= 0)
+  if ((ssize_t)output_size <= 0)
     {
```

Comment:
Why the cast to ssize_t?

Reply:
Since `output_size` is of type `size_t`, it cannot be negative. The cast corrects this on most systems, but as I already pointed out, `ssize_t` is not the appropriate type here, as it was at one point unsigned on Windows, and POSIX only requires that it can represent -1..SSIZE_MAX.
I think it would be more appropriate to change `output_type` back to `int`. That was changed back in de60ceb738327cf6d5875531bec9f4889bd5e5db but may need revisiting.

Reply:
Ah, yes, you're right - I agree it would be better to change these back to `int`s. Not sure what I was thinking there.